lib/FreeRTOS-Plus-POSIX/source/FreeRTOS_POSIX_pthread_mutex.c

```diff
@@ -297,6 +297,10 @@ int pthread_mutex_unlock( pthread_mutex_t * mutex )
 
     if( iStatus == 0 )
     {
+        /* Suspend the sheduler so that
+         * mutex is unlocked AND owner is updated atomically */
+        vTaskSuspendAll();
```

Comment:
Could you try to put a mutex here instead? suspendall is violent

Reply:
vTaskSuspendAll( ) is less invasive than disabling interrupts and also is the only other native way in FreeRTOS to use a critical section. Adding a mutex to guard the state would be more expensive in terms of RAM, so I would prefer to go ahead with this fix for now, and reconsider later, eventually, after applying the other fixes for POSIX layer.